bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/handler/ModbusDataThingHandler.java

```diff
@@ -805,7 +820,13 @@ public synchronized void onWriteResponse(ModbusWriteRequestBlueprint request, Mo
      * @return updated channel data
      */
     private Map<ChannelUID, State> processUpdatedValue(State numericState, boolean boolValue) {
-        Map<@NonNull ChannelUID, @NonNull State> states = new HashMap<>();
+        Transformation localReadTransformation = readTransformation;
+        if (localReadTransformation == null) {
+            // We should always have transformation available if thing is initalized properly
+            logger.trace("No transformation available, aborting processUpdatedValue");
+            return new HashMap<ChannelUID, State>();
```

Comment:
Could you return a `Collection.emptyMap()` here?

Reply:
Yup. https://github.com/openhab/openhab-addons/pull/7994/commits/6a46de1e2f5329b0839c2aaaf1cceb55da22dd38